src/sinks/util/buffer/metrics.rs

```diff
@@ -47,15 +50,39 @@ impl Hash for MetricEntry {
 
 impl PartialEq for MetricEntry {
     fn eq(&self, other: &Self) -> bool {
-        let mut state = DefaultHasher::new();
-        self.hash(&mut state);
-        let hash1 = state.finish();
-
-        let mut state = DefaultHasher::new();
-        other.hash(&mut state);
-        let hash2 = state.finish();
+        self.name == other.name
```

Comment:
:+1: a comment would have helped me here understand that the only real difference between `PartialEq` on `MetricEntry` vs. `Metric` is that `MetricEntry` allows different values.

Reply:
Will do.